tests/gpu-tests/test_context_retry.py

```diff
@@ -0,0 +1,310 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import subprocess
+import tempfile
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Dict
+
+import pytest
+
+from tests.conftest import docker_rm, docker_rm_and_mkdir, docker_run
+
+
+@dataclass
+class TestConfig:
+    """Configuration for test parameters"""
+
+    num_tokens_to_generate: int = 1_000_000
+    num_samples: int = 4
+    accuracy_threshold_percent: int = 25
+    server_gpus: int = 1
+    server_nodes: int = 1
+
+
+@dataclass
+class TestEnvironment:
+    """Manages test environment setup and validation"""
+
+    model_path = os.getenv("NEMO_SKILLS_TEST_HF_MODEL")
+    model_type = os.getenv("NEMO_SKILLS_TEST_MODEL_TYPE")
+
+    def validate_environment(self):
+        """Validate required environment variables"""
+        if not self.model_path:
+            pytest.skip("Define NEMO_SKILLS_TEST_HF_MODEL to run this test", allow_module_level=True)
+        if not self.model_type:
+            pytest.skip("Define NEMO_SKILLS_TEST_MODEL_TYPE to run this test", allow_module_level=True)
+
+
+class CommandBuilder:
```

Comment:
No need to change now, but for the future, the same comment applies here as on slurm ci PR. I think this is a bit overengineered and makes it hard to add new tests and also understand what existing tests do. So generally I think for tests we should go for a simpler structure with just a command + check and duplication is not a big deal imo.

Reply:
Let me fix this in another PR. 
The test script was more vibe-coded, and vibe coding tends to overengineer :)